```
FAILED: test/ioctl/libmock-ioctl.so.p/mock.c.o 
gcc -Itest/ioctl/libmock-ioctl.so.p -Itest/ioctl -I../test/ioctl -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -fomit-frame-pointer -D_GNU_SOURCE -include internal/config.h -fPIC -MD -MQ test/ioctl/libmock-ioctl.so.p/mock.c.o -MF test/ioctl/libmock-ioctl.so.p/mock.c.o.d -o test/ioctl/libmock-ioctl.so.p/mock.c.o -c ../test/ioctl/mock.c
../test/ioctl/mock.c:117:5: error: conflicting types for ‘ioctl’; have ‘int(int,  long unsigned int, ...)’
  117 | int ioctl(int fd, unsigned long request, ...)
      |     ^~~~~
In file included from ../test/ioctl/mock.c:9:
/nix/store/qnzi64i0hccghn024klcax1nd31s578f-musl-1.2.3-dev/include/sys/ioctl.h:115:5: note: previous declaration of ‘ioctl’ with type ‘int(int,  int, ...)’
  115 | int ioctl (int, int, ...);
      |     ^~~~~
```

it appears that the mock might need a musl-specific patch...

--- a/pkgs/os-specific/linux/libnvme/default.nix
+++ b/pkgs/os-specific/linux/libnvme/default.nix
@@ -11,11 +11,12 @@
 , stdenv
 , swig
 , systemd
+, fetchpatch
 # ImportError: cannot import name 'mlog' from 'mesonbuild'
 , withDocs ? stdenv.hostPlatform.canExecute stdenv.buildPlatform
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "libnvme";
   version = "1.6";
 
@@ -24,9 +25,17 @@ stdenv.mkDerivation rec {
   src = fetchFromGitHub {
     owner = "linux-nvme";
     repo = "libnvme";
-    rev = "v${version}";
+    rev = "v${finalAttrs.version}";
     hash = "sha256-7bvjsmt16/6RycSDKIECtJ4ES7NTaspU6IMpUw0sViA=";
   };
+
+  patches = [
+    # included in next release
+    (fetchpatch {
+      url = "https://github.com/linux-nvme/libnvme/commit/ff742e792725c316ba6de0800188bf36751bd1d1.patch";
+      hash = "sha256-IUjPUBmGQC4oAKFFlBrjonqD2YdyNPC9siK4t/t2slE=";
+    })
+  ];
 
   postPatch = ''
     patchShebangs scripts
@@ -51,6 +60,7 @@ stdenv.mkDerivation rec {
 
   mesonFlags = [
     "-Ddocs=man"
+    (lib.mesonBool "tests" finalAttrs.doCheck)
     (lib.mesonBool "docs-build" withDocs)
   ];
 
@@ -58,7 +68,8 @@ stdenv.mkDerivation rec {
     export KBUILD_BUILD_TIMESTAMP="$(date -u -d @$SOURCE_DATE_EPOCH)"
   '';
 
-  doCheck = true;
+  # mocked ioctl conflicts with the musl one: https://github.com/NixOS/nixpkgs/pull/263768#issuecomment-1782877974
+  doCheck = !stdenv.hostPlatform.isMusl;
 
   meta = with lib; {
     description = "C Library for NVM Express on Linux";
@@ -67,4 +78,4 @@ stdenv.mkDerivation rec {
     license = with licenses; [ lgpl21Plus ];
     platforms = platforms.linux;
   };
-}
+})

--- a/pkgs/os-specific/linux/nvme-cli/default.nix
+++ b/pkgs/os-specific/linux/nvme-cli/default.nix
@@ -4,18 +4,19 @@
 , libnvme
 , json_c
 , zlib
+, libhugetlbfs
 , python3Packages
 }:
 
 stdenv.mkDerivation rec {
   pname = "nvme-cli";
-  version = "2.4";
+  version = "2.6";
 
   src = fetchFromGitHub {
     owner = "linux-nvme";
     repo = "nvme-cli";
     rev = "v${version}";
-    hash = "sha256-vnhvVVfEDnmEIdIkfTRoiLB7dZ1rJV3U4PmUUoaxTUs=";
+    hash = "sha256-MFyBkwTNOBQdHWj7In1OquRIAsjsd4/DHYfUyFA9YDQ=";
   };
 
   mesonFlags = [
@@ -32,6 +33,8 @@ stdenv.mkDerivation rec {
     libnvme
     json_c
     zlib
+  ] ++ lib.optionals (!(stdenv.hostPlatform.isStatic || stdenv.hostPlatform.isMusl)) [
+    libhugetlbfs
   ];
 
   meta = with lib; {
@@ -47,7 +50,7 @@ stdenv.mkDerivation rec {
     '';
     license = licenses.gpl2Plus;
     platforms = platforms.linux;
-    maintainers = with maintainers; [ mic92 ];
+    maintainers = with maintainers; [ mic92 vifino ];
     mainProgram = "nvme";
   };
 }

